SOURCES/etc/xapi.d/plugins/smartctl.py

```
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import json
+import sys
+import subprocess
+import XenAPIPlugin
+
+sys.path.append('.')
+from xcpngutils import configure_logging, run_command, error_wrapped
+from xcpngutils.operationlocker import OperationLocker
+
+_LOGGER = configure_logging('smartctl')
+
+def list_disks():
+    disks = []
+    try:
+        output = subprocess.check_output(['sudo', 'smartctl', '--scan'], universal_newlines=True)
+        lines = output.split('\n')
+        for line in lines:
+            if line.startswith('/dev/') and not line.startswith('/dev/bus/'):
+                disks.append(line.split()[0])
+    except subprocess.CalledProcessError:
+        print('Error: smartctl command failed.')
```

I think we should re-raise the exception here to not mask the error.